src/Simplify.elm

```diff
@@ -1378,13 +1378,63 @@ expressionVisitorHelp node context =
                 Expression.LetExpression { expression } ->
                     onlyErrors (recordAccessLetInChecks (Node.range node) (Node.range expression))
 
+                Expression.IfBlock _ (Node thenRange thenValue) (Node elseRange elseValue) ->
+                    let
+                        nodeRange : Range
+                        nodeRange =
+                            Node.range node
+                    in
+                    if leavesAreRecords thenValue && leavesAreRecords elseValue then
```

Comment:
One thing that you can do is to, instead of returning a boolean in `leavesAreRecords`, return the ranges of the found records, maybe in the shape of `Maybe (List (Node Range))`, returning `Nothing` when one the leaves isn't a record.

That way, you can then use those records to insert the record access

```suggestion
                    case leavesAreRecords [ thenValue, elseValue ] 
                      Just ranges ->
                        onlyErrors [ Rule.error ...
                           ( Fix.removeRange { start = (Node.range record).end, end = nodeRange.end } :: (List.concatMap <insert record access> ranges))
```

(rough draft, clean it up as you see fit)

Reply:
Oh, yes, that makes a lot of sense, so we don't require multiple passes!